### PR TITLE
[infra] proposing detailed Multistage Dockerfile

### PR DESCRIPTION
**Pull request by sw360cab**
A proposal for a much detailed multistaged Dockerfile.

* it builds faster since it leverages caches
* it adds only the strictly necessary files (context copies to docker image is faster)

Further improvements:

* replace Binary base image to `scratch` from `alpine`

The latter should be evaluated carefully since such base image would not enable any access, neither attaching a terminal

### CODE REVIEW

File: Dockerfile.release
Comment (thehowl) [inline Dockerfile.release:1]: I wonder why we have this to begin with... couldn't the release version just be the Dockerfile one?

Seems sane to have either alpine or busybox bundled.

**Review by thehowl — APPROVED** [review]

**Review by ajnavarro — APPROVED** [review]

File: Dockerfile.release
Comment (ajnavarro) [reply Dockerfile.release:1]: Goreleaser needs a Dockerfile that is just copying the binary, not building it, that's why we need two Dockerfiles.

File: Dockerfile [hunk 1]
```diff
@@ -2,20 +2,31 @@
...
 #===============
 # Stage 2: Run
 #===============
 
-FROM alpine
+FROM alpine:3.20 AS tx-indexer
 
-COPY --from=builder /app/indexer /usr/local/bin/indexer
+# Set the timezone and install CA certificates
+# RUN apk --no-cache add ca-certificates tzdata
```
Comment (zivkovicmilos) [inline Dockerfile:29]: Leftover?

**Review by zivkovicmilos — APPROVED** [review]
🔥

**Pull request merged**